### DISCUSSION TITLE
Possible to ship my application with a local copy of the .NET 7 runtime?

### DISCUSSION THREAD

**Discussion by UweKeim** [dotnet/runtime#85826] [Q&A]
Having a set of multiple .NET 7 applications (Console and WinForms) that all sit in the very same folder of a product of mine, I try to find a way to deploy it.

Ideally I do not want the .NET 7 Desktop runtime to be installed as a prerequisite condition. I also do not want to publish every of my multiple .NET applications as "Self-contained", since this would enlarge the whole product deployment package as everyone of the several applications has its own copy of .NET 7.

My ideal solution would be to have all my .NET applications sit in one folder and additionally have the full .NET Desktop runtime and all of its files also sit in that very same folder so that all my .NET applications do use this local copy of the .NET Desktop runtime.

I've read through "[.NET application publishing overview](https://learn.microsoft.com/en-us/dotnet/core/deploying/)" but still found no way to do this.

### My question

Is it possible to "side-by-side" deploy the .NET 7 runtime together with my application(s)?

**Comment by UweKeim** [comment]
I wasn't fully aware that "Self-contained" and "Produce single file" are two independent switches.

I do think turning "Self-contained" on but leaving "Produce single file" off for one of my multiple applications could be sufficient to serve as the runtime for _all_ of my distinct applications in the final folder.

**Reply by UweKeim** [reply]
I've also [asked a similar question over at Stack Overflow](https://stackoverflow.com/q/76182020/107625). Maybe the answers there are helpful for anyone reading this here in the future.

**Comment by Clockwork-Muse** [comment]
You can have installers check if the runtime is already installed, and install it if it isn't (there are existing tools for this, especially on Windows).  Would that also be an option for you?

**Reply by UweKeim** [reply]
Thanks, @Clockwork-Muse. I want to avoid the user to need administrative permissions to install my application.

I'll give your suggestion a try though, thank you very much 🙂

**Comment by huoyaoyuan** [comment]
See #52974 and #53834
